pkgs/applications/audio/fluidsynth/default.nix

```diff
@@ -1,17 +1,32 @@
 { stdenv, lib, fetchFromGitHub, pkgconfig, cmake
 , alsaLib, glib, libjack2, libsndfile, libpulseaudio
 , AudioUnit, CoreAudio, CoreMIDI, CoreServices
+, version ? "2"
 }:
 
+let
+  versionMap = {
+    "1" = {
+      fluidsynthVersion = "1.1.11";
+      sha256 = "0n75jq3xgq46hfmjkaaxz3gic77shs4fzajq40c8gk043i84xbdh";
+    };
+    "2" = {
+      fluidsynthVersion = "2.0.1";
+      sha256 = "1mqyym5qkh8xd1rqj3yhfxbw5dxjcrljb6nkfqzvcarlv4h6rjn7";
+    };
+  };
+in
+
+with versionMap.${version};
+
 stdenv.mkDerivation  rec {
-  name = "fluidsynth-${version}";
-  version = "2.0.1";
```

Comment:
Actually, it would be good to leave the fluidsynth 2.x derivation unchanged, thus avoiding a mass-rebuild. You'd just need to add the version line back as `version = fluidsynthVersion;`.